driver/avgpool_driver.hpp

```diff
@@ -0,0 +1,589 @@
+    if(SetTensorNd(inputDesc, in_dim, in_stride, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing input tensor: " + inflags.GetValueStr("input_dims") + ".");
+    if(SetTensorNd(outputDesc, out_dim, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing output tensor: " + inflags.GetValueStr("output_dims") + ".");
+    if(SetTensorNd(outputGradDesc, out_dim, out_grad_stride, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing output grad tensor: " + inflags.GetValueStr("output_dims") +
+                     ".");
+    if(SetTensorNd(inputGradDesc, in_dim, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing input grad tensor: " + inflags.GetValueStr("input_dims") + ".");
+
+    return miopenStatusSuccess;
+}
+
+// Equivalent to: tensor.tranpose(0, -1).contiguous().tranpose(0, -1) incase contiguous = False
+template <typename Tgpu, typename Tref>
+std::vector<uint64_t> AvgPoolDriver<Tgpu, Tref>::ComputeStrides(std::vector<uint64_t> inputDim)
+{
+    if(!isContiguous)
+        std::swap(inputDim.front(), inputDim.back());
+    std::vector<uint64_t> strides(inputDim.size());
+    strides.back() = 1;
+    for(int i = inputDim.size() - 2; i >= 0; --i)
+        strides[i] = strides[i + 1] * inputDim[i + 1];
+    if(!isContiguous)
+        std::swap(strides.front(), strides.back());
+    return strides;
+}
+
+template <typename Tgpu, typename Tref>
+int AvgPoolDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward AvgPool (Default=1)", "int");
+    inflags.AddTensorFlag(
+        "input_dims",
+        'D',
+        "2x3x7x9",
+        "The dimensional lengths of the input tensor: N,C,D1,D2,... Example: 2x3x7x9.");
+    inflags.AddTensorFlag(
+        "kernel_size", 'k', "1x1", "The size of the window D1,D2,... Example: 1x1.");
+    inflags.AddTensorFlag(
+        "stride",
+        's',
+        "1x1",
+        "The stride of the window. Default value is kernel_size D1,D2,... Example: 1x1.");
+    inflags.AddTensorFlag(
+        "padding",
+        'p',
+        "0x0",
+        "Implicit zero padding to be added on both sides D1,D2,... Example: 0x0.");
+    inflags.AddInputFlag("ceil_mode",
+                         'c',
+                         "1",
+                         "When 1, will use ceil instead of floor to compute the output shape.",
+                         "int");
+    inflags.AddInputFlag("count_include_pad",
+                         'P',
+                         "0",
+                         "When 1, will include the zero-padding in the averaging calculation.",
+                         "int");
+    inflags.AddInputFlag("divisor_override",
+                         'd',
+                         "0",
+                         "If specified, it will be used as divisor, otherwise size of the pooling "
+                         "region will be used.",
+                         "int");
```

Comment:
Please add default value of those flags to their descriptions

Reply:
Resolved